spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/procedures/ProcedureInput.java

```diff
@@ -62,31 +62,43 @@ public boolean isProvided(ProcedureParameter param) {
     return !args.isNullAt(ordinal);
   }
 
-  public boolean bool(ProcedureParameter param, boolean defaultValue) {
+  public Boolean asBoolean(ProcedureParameter param, Boolean defaultValue) {
```

Comment:
I overlooked initially that we won't be able to have reasonable naming for retrieving longs (long is reserved). I switched to asXXX like in PropertyUtil and renamed all existing methods.